### PR TITLE
ROX-20713: Add log with connection details (External Entities in NWG)

### PR DESCRIPTION
**Pull request by vikin91**
## Description

Currently, if customers see an edge connecting a container with External Entities on the network graph, then they have no  information about this connection.

With this PR, we are adding a log line to Sensor with information about all external connections that are unknown. It means that we show source and destination of the connection (namespace + container name on one side, and IP address and port on the other side) for all cases when "External Entities" would appear on the graph.

Note that we will not show this log line, if the source/destination is external but belongs to the known CIDR ranges. In that case, we will show the name associated with the CIDR range on the graph and no log entry will be produced.

## Checklist
- [x] Investigated and inspected CI test results

### N/A
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

### Here I tell how I validated my change

- I have deployed a cluster and triggered an edge to "External Entities" (stackrox has such edges by default)
- Opened Sensor logs and visually confirmed that the log line appears.

```
(...) Warn: Outgoing connection from container stackrox/central to "169.254.169.254:80". Marking it as 'External Entities' in the network graph.
(...) Warn: Outgoing connection from container stackrox/central to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
(...) Warn: Incoming connection to container stackrox/central from "10.74.41.227:8443". Marking it as 'External Entities' in the network graph.
```

Screenshot 2023-11-10 at 12 27 36

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by ovalenti — COMMENTED** [review]
As I understand it, the new log statements can display something else than  `255.255.255.255` or `ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff` only when collector is configured with `ROX_ENABLE_EXTERNAL_IPS`. Is this correct ?

**Review by Maddosaurus — APPROVED** [review]
LGTM!

Regarding the log output: 
I don't know about v6 compatibility at this point, but we should maybe think about it as we investigate the NetworkGraph further.
Looking at [this KCS](https://access.redhat.com/solutions/5982721), it seems OpenShift can run a v4/v6 DualStack, but runs v4 only by default.

File: sensor/common/networkflow/manager/manager_impl.go [hunk 1]
```diff
@@ -466,13 +467,22 @@ func (m *networkFlowManager) enrichConnection(conn *connection, status *connStat
...
-			// Fake a lookup result.
+			// Fake a lookup result. This shows "External Entities" in the network graph
 			lookupResults = []clusterentities.LookupResult{
 				{
 					Entity:         networkgraph.InternetEntity(),
 					ContainerPorts: []uint16{port},
 				},
 			}
+			if conn.incoming {
+				log.Warnf("Incoming connection to container %s/%s from %q. "+
```
[the remark this answers is not in the document]
Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go:478]: I deployed Collector without any modifications to obtain this result. Is `ROX_ENABLE_EXTERNAL_IPS` enabled by default?

[the remark this answers is not in the document]
Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go:478]: Regarding `Warnf` and `Errorf` - this is not an error, so I didn't want to make it one. For me this classifies either as Info or as Warning, so that the customer doesn't have to enable debug logging to see this.

[the remark this answers is not in the document]
Comment (ovalenti) [reply sensor/common/networkflow/manager/manager_impl.go:478]: > I deployed Collector without any modifications to obtain this result. Is ROX_ENABLE_EXTERNAL_IPS enabled by default?

`ROX_ENABLE_EXTERNL_IPS` has been merged some time ago and is available in current stackrox master, but is disabled by default. There is something I don't grasp here, since in the disabled case, Collector is supposed to turn any external IP (which is not a known CIDR) into the "external entity address" which is 255.255.255.255

> Regarding Warnf and Errorf - this is not an error, so I didn't want to make it one. For me this classifies either as Info or as Warning, so that the customer doesn't have to enable debug logging to see this.

Yes, sorry for my lack of clarity; let me rephrase: let's suppose that there are so many connections being logged that Sensor is generating too much log content. Can we limit the processing around this new message formatting by setting the error level to Error ?

[the remark this answers is not in the document]
Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go:478]: I have just confirmed it, the IPs are visible with the flag disabled. Here is a selection of sensor logs shortly after starting:

Some show indeed `255.255.255.255:port` as a target. Do you think I should filter those out or maybe log some different text?

```
sensor-57fd746b5f-bqmvc No certificates found in /usr/local/share/ca-certificates
sensor-57fd746b5f-bqmvc No certificates found in /etc/pki/injected-ca-trust
sensor-57fd746b5f-bqmvc main: 2023/11/13 16:23:04.639283 main.go:32: Info: Running StackRox Version: 4.3.x-132-g8db41cfedb
sensor-57fd746b5f-bqmvc pkg/metrics: 2023/11/13 16:23:04.748289 server.go:142: Warn: Secure metrics server is disabled
sensor-57fd746b5f-bqmvc pkg/metrics: 2023/11/13 16:23:04.748500 server.go:142: Warn: Secure metrics server is disabled
sensor-57fd746b5f-bqmvc pkg/metrics: 2023/11/13 16:23:04.748594 server.go:142: Warn: Secure metrics server is disabled
sensor-57fd746b5f-bqmvc kubernetes/sensor: 2023/11/13 16:23:04.750638 sensor.go:65: Info: Running sensor with Kubernetes re-sync disabled
sensor-57fd746b5f-bqmvc kubernetes/sensor: 2023/11/13 16:23:04.751880 sensor.go:83: Info: Loaded Helm cluster configuration with fingerprint "dd84d249c7274030cfe945af6d168dd1c66d4507a9d9d2cd0d3a3ae579650511"
sensor-57fd746b5f-bqmvc kubernetes/sensor: 2023/11/13 16:23:04.773531 sensor.go:103: Info: Determined deployment identification: {
sensor-57fd746b5f-bqmvc   "systemNamespaceId": "74755f7d-c3ea-4c0f-8f8b-3763775dda4d",
sensor-57fd746b5f-bqmvc   "defaultNamespaceId": "d98df182-d09a-46e0-a9f8-db950e64df7d",
sensor-57fd746b5f-bqmvc   "appNamespace": "stackrox",
sensor-57fd746b5f-bqmvc   "appNamespaceId": "634f7595-6844-49ee-8c31-a345d78ca3ed",
sensor-57fd746b5f-bqmvc   "appServiceaccountId": "",
sensor-57fd746b5f-bqmvc   "k8sNodeName": "gke-piotr-11-13-nwg-default-pool-364fc60b-xlpc"
sensor-57fd746b5f-bqmvc }
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:04.782185 listener.go:49: Warn: ECR credential manager is not available: node provider is not AWS: gce://acs-team-temp-dev/us-central1-a/gke-piotr-11-13-nwg-default-pool-364fc60b-xlpc
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.782781 sensor.go:160: Info: Connecting to Central server central.stackrox:443
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.783257 sensor.go:240: Info: API services registered
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.783347 sensor.go:263: Info: All components have started
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.783369 sensor.go:269: Info: Running Sensor with connection retry: preventing sensor restart on disconnect
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.783466 sensor.go:391: Info: Attempting connection setup (client reconciliation = false)
sensor-57fd746b5f-bqmvc pkg/grpc: 2023/11/13 16:23:04.784285 server.go:283: Info: Launching backend gRPC listener
sensor-57fd746b5f-bqmvc pkg/grpc: 2023/11/13 16:23:04.784568 server.go:283: Info: Launching backend gRPC listener
sensor-57fd746b5f-bqmvc pkg/grpc: 2023/11/13 16:23:04.786109 server.go:457: Info: TLS-enabled multiplexed HTTP/gRPC server listening on :8443
sensor-57fd746b5f-bqmvc pkg/grpc: 2023/11/13 16:23:04.786659 server.go:406: Warn: failed to register Prometheus collector: descriptor Desc{fqName: "http_incoming_in_flight_requests", help: "Number of http requests which are currently running.", constLabels: {path="/v2/"}, variableLabels: {}} already exists with the same fully-qualified name and const label values
sensor-57fd746b5f-bqmvc pkg/grpc: 2023/11/13 16:23:04.786936 server.go:457: Info: TLS-enabled HTTP server listening on :9443
sensor-57fd746b5f-bqmvc common/centralclient: 2023/11/13 16:23:04.803001 grpc_connection.go:109: Info: Did not add central CA cert to gRPC connection
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.803731 sensor.go:357: Info: Updating Sensor State to: central-reachable
sensor-57fd746b5f-bqmvc kubernetes/eventpipeline: 2023/11/13 16:23:04.803775 pipeline_impl.go:124: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc kubernetes/eventpipeline: 2023/11/13 16:23:04.803800 pipeline_impl.go:129: Info: Connection established: Starting Kubernetes listener
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:04.803838 manager_impl.go:271: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/signal: 2023/11/13 16:23:04.803875 signal_service.go:54: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/processsignal: 2023/11/13 16:23:04.803895 pipeline.go:91: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/admissioncontroller: 2023/11/13 16:23:04.803910 alert_handler.go:41: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:04.803931 auditlog_manager_impl.go:55: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/image: 2023/11/13 16:23:04.803945 service_impl.go:137: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:04.803959 node_inventory_handler_impl.go:79: Info: Component runs now in Online mode
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.803972 notifiable.go:26: Info: Component 'Kernel probe server handler' runs now in Online mode
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.803983 notifiable.go:26: Info: Component 'Kernel object cache' runs now in Online mode
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:04.804023 central_communication_impl.go:151: Info: Sensor has client reconciliation disabled
sensor-57fd746b5f-bqmvc common/clusterid: 2023/11/13 16:23:04.804838 cluster_id.go:35: Info: Certificate has wildcard subject 00000000-0000-0000-0000-000000000000. Waiting to receive cluster ID from central...
sensor-57fd746b5f-bqmvc common/clusterid: 2023/11/13 16:23:04.940990 cluster_id.go:61: Info: Received dynamic cluster ID "ec47052c-915e-423d-989d-f9b42fc18feb"
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.005101 resource_event_handler.go:88: Error: Checking API resources for group "compliance.openshift.io/v1alpha1": the server could not find the requested resource
sensor-57fd746b5f-bqmvc common/config: 2023/11/13 16:23:05.043496 handler.go:89: Info: Received configuration from Central: {
sensor-57fd746b5f-bqmvc   "config": {
sensor-57fd746b5f-bqmvc     "admissionControllerConfig": {
sensor-57fd746b5f-bqmvc       "enabled": false,
sensor-57fd746b5f-bqmvc       "timeoutSeconds": 20,
sensor-57fd746b5f-bqmvc       "scanInline": false,
sensor-57fd746b5f-bqmvc       "disableBypass": false,
sensor-57fd746b5f-bqmvc       "enforceOnUpdates": false
sensor-57fd746b5f-bqmvc     },
sensor-57fd746b5f-bqmvc     "registryOverride": "",
sensor-57fd746b5f-bqmvc     "disableAuditLogs": true
sensor-57fd746b5f-bqmvc   }
sensor-57fd746b5f-bqmvc }
sensor-57fd746b5f-bqmvc common/config: 2023/11/13 16:23:05.043566 handler.go:98: Info: Stopping audit log collection
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.149761 resource_event_handler.go:192: Info: Successfully synced secrets, service accounts and roles
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:05.222743 central_communication_impl.go:283: Info: Skipping client reconciliation. Sensor will not wait for deduper state
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:05.222822 central_communication_impl.go:190: Info: Established connection to Central.
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:05.222851 central_communication_impl.go:207: Info: Communication with central started.
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.264865 resource_event_handler.go:207: Info: Successfully synced role bindings
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.264964 resource_event_handler.go:217: Info: Successfully synced k8s pod cache
sensor-57fd746b5f-bqmvc common/config: 2023/11/13 16:23:05.348171 handler.go:78: Info: Received audit log sync state from Central: {
sensor-57fd746b5f-bqmvc   "nodeAuditLogFileStates": {
sensor-57fd746b5f-bqmvc   }
sensor-57fd746b5f-bqmvc }
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.766743 resource_event_handler.go:247: Info: Successfully synced network policies, nodes, services, jobs, replica sets, and replication controllers
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.934786 resource_event_handler.go:272: Info: Successfully synced daemonsets, deployments, stateful sets and cronjobs
sensor-57fd746b5f-bqmvc kubernetes/listener: 2023/11/13 16:23:05.957008 resource_event_handler.go:281: Info: Successfully synced pods
sensor-57fd746b5f-bqmvc common/sensor: 2023/11/13 16:23:06.124922 central_sender_impl.go:115: Info: Sending synced signal to Central
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:14.549058 service_impl.go:133: Info: Received connection from "gke-piotr-11-13-nwg-default-pool-364fc60b-xlpc"
sensor-57fd746b5f-bqmvc pkg/kocache: 2023/11/13 16:23:18.714796 source.go:64: Info: loading probe 2.7.0/collector-ebpf-5.15.0-1036-gke.o.gz from upstream /kernel-objects
sensor-57fd746b5f-bqmvc pkg/kocache: 2023/11/13 16:23:20.032976 source.go:64: Info: loading probe 2.7.0/collector-ebpf-5.15.0-1036-gke.o.gz from upstream /kernel-objects
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:21.694012 service_impl.go:133: Info: Received connection from "gke-piotr-11-13-nwg-default-pool-364fc60b-gc6r"
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:29.767176 service_impl.go:133: Info: Received connection from "gke-piotr-11-13-nwg-default-pool-364fc60b-0kdl"
sensor-57fd746b5f-bqmvc common/compliance: 2023/11/13 16:23:31.681075 service_impl.go:167: Error: error receiving from compliance "gke-piotr-11-13-nwg-default-pool-364fc60b-xlpc": rpc error: code = Canceled desc = context canceled
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.805022 manager_impl.go:478: Warn: Incoming connection to container kube-system/calico-typha from "10.16.132.116:5473". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.805209 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.805370 manager_impl.go:478: Warn: Incoming connection to container kube-system/konnectivity-agent from "10.16.132.117:8093". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.805712 manager_impl.go:482: Warn: Outgoing connection from container kube-system/kubedns to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.805904 manager_impl.go:478: Warn: Incoming connection to container kube-system/kubedns from "10.16.132.117:10055". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806009 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806088 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806155 manager_impl.go:482: Warn: Outgoing connection from container kube-system/calico-typha to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806237 manager_impl.go:478: Warn: Incoming connection to container gmp-system/config-reloader from "10.16.132.117:19091". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806285 manager_impl.go:482: Warn: Outgoing connection from container kube-system/konnectivity-agent to "10.16.132.115:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806329 manager_impl.go:482: Warn: Outgoing connection from container kube-system/prometheus-to-sd to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806387 manager_impl.go:478: Warn: Incoming connection to container kube-system/sidecar from "10.16.132.117:10054". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806463 manager_impl.go:482: Warn: Outgoing connection from container kube-system/fluentbit-gke to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806511 manager_impl.go:482: Warn: Outgoing connection from container kube-system/konnectivity-agent to "10.16.132.114:8132". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806597 manager_impl.go:482: Warn: Outgoing connection from container stackrox/central to "169.254.169.254:80". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806696 manager_impl.go:482: Warn: Outgoing connection from container stackrox/central to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806783 manager_impl.go:478: Warn: Incoming connection to container kube-system/calico-typha from "10.16.132.117:5473". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806866 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "10.16.132.117:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.806972 manager_impl.go:482: Warn: Outgoing connection from container kube-system/kube-proxy to "10.16.132.114:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807031 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "10.16.132.117:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807129 manager_impl.go:478: Warn: Incoming connection to container kube-system/fluentbit-gke from "10.16.132.117:2021". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807195 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807265 manager_impl.go:478: Warn: Incoming connection to container gmp-system/prometheus from "10.16.132.117:19090". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807351 manager_impl.go:478: Warn: Incoming connection to container kube-system/calico-typha from "10.16.132.115:5473". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807428 manager_impl.go:482: Warn: Outgoing connection from container stackrox/admission-control to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807505 manager_impl.go:482: Warn: Outgoing connection from container kube-system/fluentbit-gke to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807615 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807696 manager_impl.go:478: Warn: Incoming connection to container gmp-system/config-reloader from "10.16.132.116:19091". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.807992 manager_impl.go:482: Warn: Outgoing connection from container kube-system/prometheus-to-sd to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808124 manager_impl.go:478: Warn: Incoming connection to container kube-system/sidecar from "10.16.132.116:10054". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808231 manager_impl.go:482: Warn: Outgoing connection from container kube-system/autoscaler to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808344 manager_impl.go:482: Warn: Outgoing connection from container kube-system/kubedns to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808398 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "10.16.132.116:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808453 manager_impl.go:482: Warn: Outgoing connection from container kube-system/autoscaler to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808501 manager_impl.go:482: Warn: Outgoing connection from container kube-system/gke-metrics-agent to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808545 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808594 manager_impl.go:482: Warn: Outgoing connection from container kube-system/konnectivity-agent to "10.16.132.114:8132". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808646 manager_impl.go:478: Warn: Incoming connection to container gmp-system/config-reloader from "10.16.132.116:19091". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808695 manager_impl.go:482: Warn: Outgoing connection from container kube-system/konnectivity-agent to "10.16.132.117:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808765 manager_impl.go:482: Warn: Outgoing connection from container kube-system/fluentbit-gke to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808822 manager_impl.go:482: Warn: Outgoing connection from container kube-system/kube-proxy to "10.16.132.114:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808868 manager_impl.go:478: Warn: Incoming connection to container gmp-system/prometheus from "10.16.132.116:19090". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.808918 manager_impl.go:482: Warn: Outgoing connection from container gmp-system/operator to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809013 manager_impl.go:478: Warn: Incoming connection to container kube-system/konnectivity-agent from "10.16.132.116:8093". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809155 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "10.16.132.116:10250". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809208 manager_impl.go:482: Warn: Outgoing connection from container kube-system/core-metrics-exporter to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809250 manager_impl.go:482: Warn: Outgoing connection from container kube-system/event-exporter to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809299 manager_impl.go:482: Warn: Outgoing connection from container kube-system/fluentbit-gke to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809349 manager_impl.go:478: Warn: Incoming connection to container gmp-system/operator from "10.16.132.116:18080". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809402 manager_impl.go:482: Warn: Outgoing connection from container kube-system/autoscaler to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809468 manager_impl.go:482: Warn: Outgoing connection from container kube-system/prometheus-to-sd-exporter to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809507 manager_impl.go:482: Warn: Outgoing connection from container kube-system/event-exporter to "255.255.255.255:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809554 manager_impl.go:482: Warn: Outgoing connection from container kube-system/autoscaler to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809592 manager_impl.go:482: Warn: Outgoing connection from container kube-system/autoscaler to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809645 manager_impl.go:482: Warn: Outgoing connection from container stackrox/admission-control to "10.16.133.1:443". Marking it as 'External Entities' in the network graph.
sensor-57fd746b5f-bqmvc common/networkflow/manager: 2023/11/13 16:23:34.809728 manager_impl.go:478: Warn: Incoming connection to container kube-system/kubedns from "10.16.132.116:10055". Marking it as 'External Entities' in the network graph.
```

**Comment by vikin91** [comment]
@Maddosaurus IPv6 is a good point, but I have not tested it on a cluster that would use it. I guess that for such cases, we would use the same structure:

```
type connection struct {
	local       net.NetworkPeerID
	remote      net.NumericEndpoint
	containerID string
	incoming    bool
}
```

The `net.NumericEndpoint` holds inside an `IPAddress` which `// IPAddress represents an IP (v4 or v6) address`.
I hope that this should be enough to cover both types of addresses.

File: sensor/common/networkflow/manager/manager_impl.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go]: After some consideration, I found those messages too frequent and too chatty. I changed the level of that log message to debug in [d52711c](https://github.com/stackrox/stackrox/pull/8580/commits/d52711c0ef39a1112134434a2523fdde795f89c3)

**Comment by vikin91** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8580** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8580?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #8648** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8648?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/8580?utm_source=stack-comment).

File: sensor/common/networkflow/manager/manager_impl.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ovalenti) [reply sensor/common/networkflow/manager/manager_impl.go]: Thank you for verifying with ROX_ENABLE_EXTERNAL_IPS. Indeed, the addresses here are already grouped for the public range. If we enable external IPs in the future, there will be much more data flowing, I am afraid.

**Review by ovalenti — APPROVED** [review]
Looks fine to me.

I am just a bit concerned with the processing involved in formatting the strings. Lazy evaluation mechanism in Go seems to be planned for Go2 (https://github.com/golang/go/issues/37739).
IP conversion to string and concatenation will seemingly always occur whatever the log level (https://github.com/stackrox/zap/blob/master/sugar.go#L218)

File: sensor/common/networkflow/manager/manager_impl.go [hunk 2]
```diff
@@ -466,13 +467,22 @@ func (m *networkFlowManager) enrichConnection(conn *connection, status *connStat
...
 			lookupResults = []clusterentities.LookupResult{
 				{
 					Entity:         networkgraph.InternetEntity(),
 					ContainerPorts: []uint16{port},
 				},
 			}
+			if conn.incoming {
+				log.Debugf("Incoming connection to container %s/%s from %q. "+
+					"Marking it as 'External Entities' in the network graph.",
+					container.Namespace, container.ContainerName, conn.remote.IPAndPort.String()+":"+strconv.Itoa(int(port)))
```
Comment (ovalenti) [inline sensor/common/networkflow/manager/manager_impl.go:480]:
```suggestion
				log.Debugf("Incoming connection to container %s/%s from %q:%q. "+
					"Marking it as 'External Entities' in the network graph.",
					container.Namespace, container.ContainerName, conn.remote.IPAndPort.String(), strconv.Itoa(int(port)))
```

Would save a few cycles and memory allocation when the log is disabled.

Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go:480]: Good point, but I must change `%q` to `%s`, otherwise the log would be `... from "10.0.0.1":"443".` as `%q` is quoting the element.

Comment (vikin91) [reply sensor/common/networkflow/manager/manager_impl.go]: Got that covered in [b66d6cf](https://github.com/stackrox/stackrox/pull/8580/commits/b66d6cf868612ecc2431b945b690cf5c2bdd7ee1)
